spyder_kernels/customize/spyderpdb.py

```diff
@@ -184,7 +190,30 @@ def default(self, line):
                 sys.displayhook = self.displayhook
                 if execute_events:
                      get_ipython().events.trigger('pre_execute')
-                exec(code, ns, locals)
+
+                if uses_comprehension(line):
```

Comment:
```suggestion
                 # Mitigates a CPython bug (https://bugs.python.org/issue41918)
                 # that prevents running comprehensions with the frame locals
                 # in Pdb.
                 # See https://bugs.python.org/issue21161 and
                 # spyder-ide/spyder#13909.
                if uses_comprehension(line):
```

- Move comment here so we can place the other one below the `if`.
- Improve wording.